src/lib/turtle.js

```diff
@@ -2028,18 +2046,35 @@ function generateTurtleModule(_target) {
         return result;
     }
 
-    function createColor(color, g, b, a) {
+    function createColor(turtle,color, g, b, a) {
```

Comment:
All we actually need from the turtle object here is the color mode setting so it would be better to just accept that value rather than accepting a reference to the entire turtle object.